lib/container.js

```diff
@@ -112,12 +112,28 @@ export default class Container {
     return new EventHandle(this.ee, USER_CHANGED, listener);
   }
 
-  signupWithUsername(username, password) {
-    return this._signup(username, null, password);
+  signupWithUsername(username, password, profile = null) {
+    const promise = this._signup(username, null, password);
+    if (profile === null) {
+      return promise;
+    }
+    return promise.then((user)=>
+      this._createProfile(user, profile).then(createdProfile =>
+        [user, createdProfile]
+      )
+    );
   }
 
-  signupWithEmail(email, password) {
-    return this._signup(null, email, password);
+  signupWithEmail(email, password, profile = null) {
+    const promise = this._signup(null, email, password);
+    if (profile === null) {
+      return promise;
+    }
+    return promise.then((user)=>
+      this._createProfile(user, profile).then(createdProfile =>
+        [user, createdProfile]
```

Comment:
In my opinion, the utility of the `_user` object is pretty low. How about returning a profile from `signupWithEmailAndProfile` instead of returning an array of object?